### PR TITLE
Move sitemap tests into integration and remove setup and repeated get…

### PR DESCRIPTION
**Pull request by murny**
… requests

https://github.com/ualbertalib/jupiter/issues/417

### CODE REVIEW

File: test/integration/sitemap_test.rb [hunk 1]
```diff
@@ -1,5 +1,4 @@
 require 'test_helper'
-require 'open-uri'
 
 class SitemapTest < ActionDispatch::IntegrationTest
```
Comment (murny) [inline test/integration/sitemap_test.rb]: Should have been SitemapControllerTest here if we were going to keep it the same

Reason to move to integration test? These tests are not testing the controller actions of the sitemap controller but instead are testing much more including interacting with the controller, routes and view (asserting differences in the dom/xml of the presenting page)

File: test/integration/sitemap_test.rb [hunk 2]
```diff
@@ -1,5 +1,4 @@
 require 'test_helper'
-require 'open-uri'
```
Comment (murny) [inline test/integration/sitemap_test.rb]: Never seeing this being used in this file? Removed

File: test/integration/sitemap_test.rb [hunk 3]
```diff
@@ -103,57 +98,48 @@ def before_all
...
+      assert_select 'loc', { count: 0, text: item_url(@private_item) }, 'private items should not appear in the sitemap'
     end
   end
 
   context 'collections sitemap' do
-    setup do
-      get collections_sitemap_url
-    end
     should 'be valid sitemap xml' do
       get collections_sitemap_url
```
Comment (murny) [inline test/integration/sitemap_test.rb]: Here is a bad example of how confusing the setup block can get. We are actually doing two `get collections_sitemap_url` in this should block

File: test/integration/sitemap_test.rb [hunk 4]
```diff
@@ -103,57 +98,48 @@ def before_all
         assert_select 'rs|ln[length=?]', uo.original_file.size.to_s
         assert_select 'rs|ln[type=?]', uo.original_file.mime_type.to_s
       end
-    end
 
-    should 'not show private items' do
```
Comment (murny) [inline test/integration/sitemap_test.rb]: Here and in many other places we doing a full `get request` to the page just for a couple assertions. In my opinion it's okay to group these assertions under one get request

File: app/views/profile/index.html.erb [hunk 5]
```diff
@@ -1,8 +1,6 @@
 <% page_title(@user.name) %>
 
-<div class="mb-3">
-  <h1><%= @user.name %></h1>
-</div>
+<h1 class="mb-3"><%= @user.name %></h1>
```
Comment (murny) [inline app/views/profile/index.html.erb]: Oops one of my unrelated changes snuck in here, but pretty simple. Just cleanup of some markup

File: test/integration/sitemap_test.rb (hunk 2 above)
Comment (pgwillia) [reply test/integration/sitemap_test.rb]: Good catch!  I started out using it, but moved away.

File: test/integration/sitemap_test.rb (hunk 1 above)
Comment (pgwillia) [reply test/integration/sitemap_test.rb]: It started as an integration test but I was running into problems with capybara interfering with what was returned.

File: test/integration/sitemap_test.rb (hunk 4 above)
Comment (pgwillia) [reply test/integration/sitemap_test.rb]: I was trying to organize what/how was being tested.  I'm okay with these being combined. Faster!

**Pull request merged**